keps/0032-community-repository-management.md

```diff
@@ -0,0 +1,96 @@
+- Providing multiple versions of an operator is only possible by using separate folders. Teams developing operators might prefer different workflows (e.g. Git tags)
+
+These requirements make it necessary for operator packages to be developed in a separate Git repository. As a result, the operator packages in `kudobuilder/operators` are copies of specific versions of the respective upstream Git repository. This approach has challenges as well, because bugs discovered in an operator need to be resolved in the upstream Git repository, not in `kudobuilder/operators`. I.e., there needs to be metadata to link to the upstream Git repository. By having some form of metadata describing upstream operator package sources, the same metadata can be used to describe other properties of a package, e.g. the maturity level.
+
+### Goals
+
+- Provide a simple workflow for upstream operator packages to get added to the community repository
+- Remove the need to "host" copies of upstream operator packages in a Git repository
+- Provide a mechanism to define upstream operator package sources
+- Provide a mechanism to define upstream operator package maturity
+
+### Non-Goals
+
+- Change the practice of creating a PR against a Git repository to add operator packages to the community repository
+- Change operator package management as defined in [KEP-10](0010-package-manager.md) and [KEP-15](0015-repository-management.md)
+
+## Proposal
+
+Upstream operator developers still create PRs against a Git repository to add their operator packages to the community repository. This Git repository lists references to upstream operator packages instead of a full copy of an operator package. For example, consider an operator package developed at `github.com/example/example-operator` that has a tagged version `1.0.0` and the operator package in the `operator` folder. To add or update this operator package, the developers would create a PR referencing their upstream Git repository and the specific version, e.g. by adding a file `example-operator.yaml` like
+
+```yaml
+name: Example Operator
+versions:
+- version: "1.0.0"
+  git:
+    repository: github.com/example/example-operator.git
+    tag: "1.0.0"
+    dir: operator
+```
+
+Once this PR is merged, CI tooling detects the new YAML file, clones the referenced upstream Git repository, checks out the tag, and adds the operator package in the specified folder to the existing index. Of course, CI tests that don't update the community repository can run before the PR is merged. This workflow is similar to [krew-index](https://github.com/kubernetes-sigs/krew-index)
+
+We can add more metadata to the YAML reference file. E.g., support for different upstream sources like Mercurial.
+
+The maturity level of an operator package can be provided in this file as well.
+
+Other useful metadata would be links to the operator homepage or documentation.
```

Comment:
How would these differ from the internal values to the Operator/OperatorVersion these contain.  e.g. https://github.com/kudobuilder/kudo/blob/main/pkg/apis/kudo/v1beta1/operator_types.go#L28

Reply:
We need to discuss which values make sense here versus having them in the operator package. IMO, most metadata should be provided by the operator package.

Reply:
Yeah.. I prefer metadata to be in the the tarball to be retrieved if needed by this tool.  There are things that likely make sense to have outside the bundle... still thinking

Reply:
Agreed. As mentioned [here](https://github.com/kudobuilder/kudo/pull/1574/files#r449450608), it can make sense to duplicate some metadata though, as this can help debug CI failures.